test/proposals/mips/mip-b05/mip-b05.sol

```diff
@@ -0,0 +1,304 @@
+
+contract mipb05 is Proposal, CrossChainProposal, ChainIds, Configs {
+    string public constant name = "MIP-b05";
+    uint256 public constant timestampsPerYear = 60 * 60 * 24 * 365;
+    uint256 public constant SCALE = 1e18;
+
+    struct IRParams {
+        uint256 kink;
+        uint256 baseRatePerTimestamp;
+        uint256 multiplierPerTimestamp;
+        uint256 jumpMultiplierPerTimestamp;
+    }
+
+    constructor() {
+        _setNonce(5);
+        bytes memory proposalDescription = abi.encodePacked(
+            vm.readFile("./test/proposals/mips/mip-b05/MIP-B05.md")
+        );
+        _setProposalDescription(proposalDescription);
+    }
+
+    function _validateJRM(address jrmAddress, address tokenAddress, IRParams memory params)
+        internal
+    {
+        JumpRateModel jrm = JumpRateModel(jrmAddress);
+        assertEq(
+            address(MToken(tokenAddress).interestRateModel()),
+            address(jrm),
+            "interest rate model not set correctly"
+        );
+
+        assertEq(jrm.kink(), params.kink, "kink verification failed");
+        assertEq(
+            jrm.timestampsPerYear(),
+            timestampsPerYear,
+            "timestamps per year verifiacation failed"
+        );
+        assertEq(
+            jrm.baseRatePerTimestamp(),
+            (params.baseRatePerTimestamp * SCALE) / timestampsPerYear / SCALE,
+            "base rate per timestamp validation failed"
+        );
+        assertEq(
+            jrm.multiplierPerTimestamp(),
+            (params.multiplierPerTimestamp * SCALE) / timestampsPerYear / SCALE,
+            "multiplier per timestamp validation failed"
+        );
+        assertEq(
+            jrm.jumpMultiplierPerTimestamp(),
+            (params.jumpMultiplierPerTimestamp * SCALE) /
+                timestampsPerYear /
+                SCALE,
+            "jump multiplier per timestamp validation failed"
+        );
+    }
+
+    function _validateCF(Addresses addresses, address tokenAddress, uint256 collateralFactor) internal {
+        address unitrollerAddress = addresses.getAddress("UNITROLLER");
+        Comptroller unitroller = Comptroller(unitrollerAddress);
+
+        (bool _listed, uint256 collateralFactorMantissa) = unitroller.markets(tokenAddress);
+
+        assertEq(
+            collateralFactorMantissa,
+            collateralFactor,
+            "collateral factor validation failed"
+        );
+    }
+
+    function deploy(Addresses addresses, address) public override {}
+
+    function afterDeploy(Addresses addresses, address) public override {}
+
+    function afterDeploySetup(Addresses addresses) public override {}
+
+    function build(Addresses addresses) public override {
+        address unitrollerAddress = addresses.getAddress("UNITROLLER");
+
+        // =========== ETH CF Update ============
+
+        // Check Preconditions
+        _validateCF(addresses, addresses.getAddress("MOONWELL_WETH"), 0.75e18);
```

Comment:
it's strange to add these checks in the build function. I think the pre-validations should either happen in `afterDeploy` or `afterDeploySetup`.